builtin/game/falling.lua

```diff
@@ -33,11 +33,105 @@ core.register_entity(":__builtin:falling_node", {
 				end
 			end
 		end
+		local def = core.registered_nodes[node.name]
+		if not def then
+			-- Don't allow unknown nodes to fall
+			minetest.log("warning", "Unknown falling node removed at "..minetest.pos_to_string(self.object:get_pos()))
+			self.object:remove()
+			return
+		end
 		self.meta = meta
-		self.object:set_properties({
-			is_visible = true,
-			textures = {node.name},
-		})
+		if def.drawtype == "airlike" then
+			self.object:set_properties({
+				is_visible = false,
+			})
+		else
+			if def.drawtype == "torchlike" or def.drawtype == "signlike" then
```

Comment:
`elseif def.drawtype == "torchlike" or def.drawtype == "signlike" then`
Daisy-chain style has the same efficiency and reduces the amount of indents at the same time.